ee/tasks/test/subscriptions/test_slack_subscriptions.py

```diff
@@ -0,0 +1,169 @@
+                "type": "section",
+                "text": {"type": "mrkdwn", "text": "Your subscription to the Dashboard *private dashboard* is ready!",},
+            },
+            {
+                "type": "image",
+                "image_url": post_message_calls[0].kwargs["blocks"][1]["image_url"],
+                "alt_text": "My Test subscription",
+            },
+            {"type": "section", "text": {"type": "mrkdwn", "text": "_See 🧵 for more Insights_"}},
+            {"type": "divider"},
+            {
+                "type": "actions",
+                "elements": [
+                    {
+                        "type": "button",
+                        "text": {"type": "plain_text", "text": "View in PostHog"},
+                        "url": f"http://localhost:8000/dashboard/{self.dashboard.id}?utm_source=posthog&utm_campaign=subscription_report&utm_medium=slack",
+                    },
+                    {
+                        "type": "button",
+                        "text": {"type": "plain_text", "text": "Manage Subscription"},
+                        "url": f"http://localhost:8000/dashboard/{self.dashboard.id}/subscriptions/{self.subscription.id}?utm_source=posthog&utm_campaign=subscription_report&utm_medium=slack",
+                    },
+                ],
+            },
+        ]
+
+        # Second call - other asset
+        second_call = post_message_calls[1].kwargs
+        assert second_call["channel"] == "C12345"
+        assert second_call["thread_ts"] == "1.234"
+        assert second_call["blocks"] == [
+            {"type": "image", "image_url": second_call["blocks"][0]["image_url"], "alt_text": "My Test subscription",}
+        ]
+
+        # Third call - other asset
+        third_call = post_message_calls[2].kwargs
+        assert third_call["blocks"] == [
+            {"type": "image", "image_url": third_call["blocks"][0]["image_url"], "alt_text": "My Test subscription",}
+        ]
+
+        # Fourth call - notice that more exists
+        fourth_call = post_message_calls[3].kwargs
+        assert fourth_call["blocks"] == [
+            {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": f"Showing 3 of 10 Insights. <http://localhost:8000/dashboard/{self.dashboard.id}?utm_source=posthog&utm_campaign=subscription_report&utm_medium=slack|View the rest in PostHog>",
+                },
+            }
+        ]
+
+    def test_subscription_delivery_missing_integration(self, MockSlackIntegration: MagicMock) -> None:
+        mock_slack_integration = MagicMock()
+        MockSlackIntegration.return_value = mock_slack_integration
+
+        self.integration.delete()
+
+        send_slack_subscription_report(self.subscription, [self.asset], 1)
+
+        assert mock_slack_integration.client.chat_postMessage.call_count == 0
+
+        # TODO: Should we perhaps save something to say the Subscription failed?
```

Comment:
We could `incr` a `subscriptions.success` or `subscriptions.failure` counter tagged with (insight|dashboard), (email|slack) and team id. 

Then we could alert on failures over a set threshold

Reply:
I think something like this sounds good. I'm going to make a follow up issue so we can continue with this as is for now 👍